src/ReactiveDomain.Testing/Specifications/TestQueue.cs

```diff
@@ -166,10 +181,12 @@ public void WaitForMsgId(Guid id, TimeSpan timeout)
 
                     if (Messages.ToArray().Any(m => m.MsgId == id)) { waithandle.Set(); }
 
+                    //wait here to see if the message handler triggers the wait handle we added
                     while (!waithandle.Wait(10))
                     {
                         if (DateTime.Now > deadline) { throw new TimeoutException($"Msg with ID {id} failed to arrive within {timeout}."); }
                         if (_disposed) { throw new ObjectDisposedException(nameof(TestQueue)); }
+                        if (Interlocked.Read(ref _queueVersion) != version) { throw new InvalidOperationException("Test queue Cleared!"); }
```

Comment:
This error message doesn't look quite right. If `_queueVersion` is greater than `version` it might be because something else came in since we first read the version.

Reply:
_queueVersion only increments on clear, not per message
I'm open to a better name

Reply:
Ah. Yeah, we've used "version" in other contexts to indicate the number of messages handled, so a different name would be great.

Reply:
I think the name is fine as-is, it's not in a visible part of the code, and upon examination the usage is clear.